src/aura/BGE_DataImportBatchEntry/BGE_DataImportBatchEntryHelper.js

```diff
@@ -84,46 +106,74 @@
 
     /**
      * @description: handles the display or clearing of errors from the results of dry run
-     * @param rowErrors: object with row Id, title, and list of associated messages
+     * @param rowErrors: list of objects with row Id, title, and list of associated messages
      */
-    handleTableErrors: function(component, rowErrors) {
-        var tableErrors = { rows: {}, table: {}, size: 0 };
+    handleNewTableErrors: function(component, newErrors) {
+        let tableErrors = component.get('v.errors');
+        if (!tableErrors) {
+            tableErrors = { rows: {}, table: {}, size: 0 };
+        }
 
-        rowErrors.forEach(function(error) {
-            var rowError = {
-                title: error.title,
-                messages: error.messages
-            };
-            tableErrors.rows[error.id] = rowError;
+        newErrors.forEach(function(error) {
+            //only insert new errors
+            if (!tableErrors.rows.hasOwnProperty(error.id)) {
+                const rowError = {
+                    title: error.title,
+                    messages: error.messages
+                };
+                tableErrors.rows[error.id] = rowError;
+            }
         });
 
-        tableErrors.size = rowErrors.length;
+        tableErrors.size = newErrors.length;
 
         component.set('v.errors', tableErrors);
     },
 
     /**
-     * @description: saves inline edits from dataTable.
-     * @param draftValues: changed values in the table
+     * @description: rebuilding error Object and removes any errors on the table if a row is updated or deleted
+     * @param rowId: Id of the row with the error that needs clearing
      */
-    handleTableSave: function(component, draftValues) {
+    clearOldTableErrors: function(component, rowId) {
+        let tableErrors = component.get('v.errors');
+        if (!tableErrors) {
+            tableErrors = { rows: {}, table: {}, size: 0 };
+        }
+
+        if (tableErrors.rows.hasOwnProperty(rowId)) {
+            let updatedErrors = { rows: {}, table: {}, size: 0 };
+            let errorSize = 0;
+            let errorIds = Object.keys(tableErrors.rows);
+            errorIds.forEach(function(oldErrorId) {
+                if (oldErrorId !== rowId) {
+                    updatedErrors.rows[oldErrorId] = tableErrors.rows[oldErrorId];
+                    errorSize++;
+                }
+            });
+            updatedErrors.size = errorSize;
+            component.set('v.errors', updatedErrors);
```

Comment:
Is updatedErrors just a duplicated list of errors except you aren't adding the error at rowId into it? Would it be possible to just remove the error at tableErrors.rows[rowId]?

Reply:
@ptewson-sfdo thanks for nudging me on this one, I was overthinking it. I needed to call delete on it, because rowId is an attribute on the row Object, and was having a tough time because I thought it was a function on the object!